pkgs/build-support/trivial-builders.nix

```
@@ -1,5 +1,31 @@
-{ lib, stdenv, stdenvNoCC, lndir, runtimeShell }:
+{ lib, stdenv, stdenvNoCC, lndir, runtimeShell, zsh, dash, bash}:
 
+let
+  runtime = { 
+    preamble = "#!${runtimeShell}";
+    checkPhase = ''
+      ${stdenv.shell} -n $out
```

I believe you'll want `runtimeShell` here as well:
```suggestion
      ${runtimeShell} -n $out
```

---

yup! That's what I get for writing this in 5 minutes ha.